client/ayon_shotgrid/plugins/publish/integrate_shotgrid_version.py

```diff
@@ -43,38 +43,58 @@ def process(self, instance):
         # find thumbnail path
         thumbnail_path = instance.data.get("thumbnailPath")
 
+        # set pathtoframes exclusion filters
+        frames_path_filters = ["thumb", "workfile"]
+
         found_reviewable = False
+        found_frames = False
+
         for representation in instance.data.get("representations", []):
             self.log.debug(pformat(representation))
 
-            if "shotgridreview" not in representation.get("tags", []):
+            # Skip unwanted representations
+            if any(filter_name for filter_name in frames_path_filters if filter_name in f".{representation['name']}"):
                 continue
 
             local_path = get_publish_repre_path(
                 instance, representation, False
             )
 
-            if f".{representation['ext']}" in VIDEO_EXTENSIONS:
-                found_reviewable = True
-                data_to_update["sg_path_to_movie"] = local_path
-                if (
-                    "slate" in instance.data["families"]
-                    and "slate-frame" in representation["tags"]
-                ):
-                    data_to_update["sg_movie_has_slate"] = True
-
-            elif f".{representation['ext']}" in IMAGE_EXTENSIONS:
-                found_reviewable = True
-                # Replace the frame number with '%04d'
-                path_to_frame = re.sub(r"\.\d+\.", ".%04d.", local_path)
-
-                data_to_update.update({
-                    "sg_path_to_movie": path_to_frame,
-                    "sg_path_to_frames": path_to_frame,
-                })
+            # Skip temp locations
+            if local_path.startswith("/tmp"):
+                continue
 
-                if "slate" in instance.data["families"]:
-                    data_to_update["sg_frames_have_slate"] = True
+            # Replace the frame number with '####'
+            local_path = re.sub(r"\.\d+\.", ".####.", local_path)
+
+            if "shotgridreview" not in representation.get("tags", []) and not found_frames:
```

Comment:
can you explain the logic behind this?

Reply:
If I understand the code, I think this changes a bit the logic..

What used to be:
1. inspect whatever representation was found under `shotgridreview` tag 
2. fill up `sg_path_to_movie` and/or `sg_path_to_frames`/`sg_frames_have_slate` accordingly

becomes:
1. loop over all representations. As soon as a non-temp img-seq repr. is found, use it for `sg_path_to_frames`
2. potentially override it with `shotgridreview` tagged repr. unless it was tagged `shotgridpathtoframes`

What I'm unsure with is that there does not seem to be any guaranteed order between `shotgridpathtoframes` and `shotgridreview`. It'll be depending on which representation comes first.

Maybe that should be:
1. attempt to detect `shotgridpathtoframes` and use it if found
2. otherwise try to get it from `shotgridreview` 
3. last resort, fill up with any other img-sequence matching repr.
 (_Personally I'm doubtful about this one cause I feel like if a `shotgridpathtoframes` tag feature is available, we might as well enforce it instead of providing a random default)_